test/e2e/integration_test.go

```diff
@@ -40,94 +42,126 @@ var _ = Describe("Configuration namespaced. Deploy deployment", Label("integrati
+				30018,
+				[]func(*model.TestDataProvider){},
+			).WithProject(data.DefaultProject()),
+			[]project.Integration{
+				{
+					Type:   "DATADOG",
+					Region: "EU",
+				},
+			},
+			datadogEnvKey,
+		),
+		Entry("Users can integrate PagerDuty on region US", Label("project-integration"),
+			model.DataProvider(
+				"pager-duty-us",
+				model.NewEmptyAtlasKeyType().UseDefaultFullAccess(),
+				30018,
+				[]func(*model.TestDataProvider){},
+			).WithProject(data.DefaultProject()),
+			[]project.Integration{
+				{
+					Type:   "PAGER_DUTY",
+					Region: "US",
+				},
+			},
+			datadogEnvKey,
 		),
 	)
 })
 
-func integrationCycle(data *model.TestDataProvider, key string) {
-	integrationType := "DATADOG"
+func integrationTest(data *model.TestDataProvider, integration project.Integration, key string) {
+	By("Create Secret for integration", func() {
+		Expect(k8s.CreateUserSecret(data.Context, data.K8SClient, key, integrationSecretName, data.Resources.Namespace)).Should(Succeed())
 
-	By("Deploy User Resources", func() {
-		projectStatus := GetProjectIntegrationStatus(data)
-		Expect(projectStatus).Should(BeEmpty())
+		integration.APIKeyRef = common.ResourceRefNamespaced{
+			Name:      integrationSecretName,
+			Namespace: data.Resources.Namespace,
+		}
 	})
 
 	By("Add integration", func() {
-		Expect(data.K8SClient.Get(data.Context, types.NamespacedName{Name: data.Project.Name,
-			Namespace: data.Resources.Namespace}, data.Project)).Should(Succeed())
-		newIntegration := project.Integration{
-			Type: integrationType,
-			APIKeyRef: common.ResourceRefNamespaced{
-				Name:      "datadog-secret",
-				Namespace: data.Resources.Namespace,
-			},
-			Region: "EU",
-		}
-		data.Project.Spec.Integrations = append(data.Project.Spec.Integrations, newIntegration)
-		By("Create Secret for integration", func() {
-			for _, i := range data.Project.Spec.Integrations {
-				Expect(k8s.CreateUserSecret(data.Context, data.K8SClient, key, i.APIKeyRef.Name, i.APIKeyRef.Namespace)).Should(Succeed())
-			}
-		})
+		Expect(data.K8SClient.Get(data.Context, types.NamespacedName{Name: data.Project.Name, Namespace: data.Resources.Namespace}, data.Project)).Should(Succeed())
+		data.Project.Spec.Integrations = append(data.Project.Spec.Integrations, integration)
+
 		Expect(data.K8SClient.Update(data.Context, data.Project)).Should(Succeed())
-		actions.WaitForConditionsToBecomeTrue(data, status.IntegrationReadyType, status.ReadyType)
 	})
 
-	atlasClient := atlas.GetClientOrFail()
-	By("Check statuses", func() {
-		var projectStatus string
-		projectStatus, err := k8s.GetProjectStatusCondition(data.Context, data.K8SClient, status.IntegrationReadyType, data.Resources.Namespace, data.Project.GetName())
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(projectStatus).Should(Equal("True"))
-
-		Expect(err).ShouldNot(HaveOccurred())
+	By("Integration is ready", func() {
+		actions.WaitForConditionsToBecomeTrue(data, status.IntegrationReadyType, status.ReadyType)
```

Comment:
clarification question: this waits for the Update which happened in line 142 to be reflected in the conditions, right?

Reply:
Yes, we wait the operator do his job and see the condition `IntegrationReadyType` become true